### PR TITLE
style: basic layout styling for feedback survey

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
- Basic layout styling for all survey question types
- Adds optional "subtitle" field for `info`  questions

### CODE REVIEW

File: components/RadioButtons.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const RadioButtons = ({
 
   return (
     <>
-      {title && <h3>{title}</h3>}
+      {title && <h3 className={styles.title}>{title}</h3>}
```
Comment (AdrianaCeric) [inline components/RadioButtons.tsx]: Just as heads up for a11y heading hierarchy, my PR will [change the `h3` to an `h2`](https://github.com/ibi-group/survey/pull/17/files#diff-5bc2b5087dcba0ccdfe1abf46b3613772b9d3b43eba6b8a748bc3eda588cd3c3R42-R51). 
The css h3 selectors should also be changed to h2.

**Review by AdrianaCeric — APPROVED** [review]
Changes look great! Love the minimalistic black and white look. Just realized our two PRs are working on the same file components. There are structural changes (for a11y) that I made to each title like adding a `<legend>` to each `<fieldset>` and switching h3 --> h2. This PR may be blocked until that one's merged?

File: components/QuestionRenderer.tsx [hunk 2]
```diff
@@ -45,7 +46,12 @@ const QuestionRenderer = ({
 
   switch (type) {
     case 'info':
-      return <h1>{title}</h1>
+      return (
+        <>
+          <h1>{title}</h1>
+          <span className="subtitle">{subtitle}</span>
```
Comment (miles-grant-ibigroup) [inline components/QuestionRenderer.tsx]: Can we make this an h2 instead? Then we don't need a class on it either

File: components/RadioButtons.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [inline components/RadioButtons.tsx]: The other questions now have centered question titles, but this one isn't centered. Do we want to move those centering styles to the global h2/h3 style instead?

File: components/Stars.tsx [hunk 3]
```diff
@@ -52,9 +52,9 @@ const Stars = ({
 
   return (
     <>
-      {title && <h3>{title}</h3>}
+      {title && <h3 className="center">{title}</h3>}
       <fieldset
-        className={styles.container}
+        className={`${styles.container} ${starStyles.container}`}
```
Comment (miles-grant-ibigroup) [inline components/Stars.tsx]: Clever, but why is the star container different?

File: config.yaml [hunk 4]
```diff
@@ -63,8 +69,8 @@ i18n:
     index: 
       next: "Next 〉"
       prev: "〈 Previous"
-      completion: "Thanks. Your response has been submitted."
-      completionSecondary: ""
+      completion: "Thanks!"
+      completionSecondary: "Your response has been submitted."
```
Comment (miles-grant-ibigroup) [inline config.yaml]: Can we rename this `completionSubtitle` for consistency? We'll have to make it a breaking change

File: config.yaml [hunk 5]
```diff
@@ -50,6 +51,11 @@ questions:
         options:
           - "option 1"
           - "this is an example of making a mistake and mismatching the number of options in each language. Things will render properly, but users of this langauge will not be able to select all options!"
+  -
+    # Example of stars question
+    type: stars
+    title: "How many stars do you see?"
```
Comment (miles-grant-ibigroup) [inline config.yaml:57]: Lol I like this

File: pages/index.tsx [hunk 6]
```diff
@@ -93,7 +93,7 @@ const Home: NextPage = ({ questions, socketServerUrl }: Props) => {
       {surveyOver && (
         <div className={styles.question}>
           <h1>{t('index.completion')}</h1>
-          <h2>{t('index.completionSecondary')}</h2>
+          <span className="subtitle">{t('index.completionSecondary')}</span>
```
Comment (miles-grant-ibigroup) [inline pages/index.tsx]: Why did we change this to a span?

File: styles/RadioButtons.module.css [hunk 7]
```diff
@@ -22,11 +22,12 @@
   -moz-appearance: none;
 }
 .button:focus ~ .label {
-  border-style: dashed !important;
+  outline: rgb(124, 186, 231) 2px solid;
```
Comment (miles-grant-ibigroup) [inline styles/RadioButtons.module.css]: image
Is this intentional? I think it's a bit funky

File: styles/RadioButtons.module.css [hunk 8]
```diff
@@ -22,11 +22,12 @@
...
+  outline-offset: 2px;
 }
 
 .button ~ .label {
-  padding: 3em;
+  padding: .85em 1.75em;
```
Comment (miles-grant-ibigroup) [inline styles/RadioButtons.module.css]: I like the new paddings, but I'd like to try to stick to `.5` increments if possible just to keep grids clean

File: styles/globals.css [hunk 9]
```diff
@@ -7,7 +7,6 @@ body {
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
   color: #222;
-  background: rgb(192, 230, 255);
```
Comment (miles-grant-ibigroup) [inline styles/globals.css:10]: I agree it looks cleaner with white but could we make this configurable?

File: styles/globals.css [hunk 10]
```diff
@@ -18,3 +17,21 @@ a {
...
+h3 {
+  font-size: 1.75em;
+}
+
+h3.center {
+  padding-left: 20px
+}
+
+.subtitle {
+  font-size: 24px;
```
Comment (miles-grant-ibigroup) [inline styles/globals.css]: Should we have all font sizes in `em`?

File: styles/globals.css [hunk 11]
```diff
@@ -18,3 +17,21 @@ a {
...
+h1, h3.center {
+  text-align: center;
+}
+
+h3 {
+  font-size: 1.75em;
+}
+
+h3.center {
+  padding-left: 20px
```
Comment (miles-grant-ibigroup) [inline styles/globals.css]: Why are we doing this instead of flex box? Or if you want to get really fancy putting a `display: grid` on the container

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I really like the end product I think it looks clean and I like how the new layout looks. Just some nitpicks on the implementation!

Also I think we should consider gridding the radio buttons/entire container -- I don't like the current reflow setup. I think they should all be in 1 row and if that doesn't fit one per row. This is easy to accomplish with `minmax` and grid

File: components/RadioButtons.tsx (hunk 1 above)
Comment (AdrianaCeric) [reply components/RadioButtons.tsx]: That sounds like a good idea

File: components/Stars.tsx (hunk 3 above)
Comment (amy-corson-ibigroup) [reply components/Stars.tsx]: It was because they used to share `display` styles! No longer relevant though.

File: pages/index.tsx (hunk 6 above)
Comment (amy-corson-ibigroup) [reply pages/index.tsx]: To match the first subtitle! I've changed both to `h2`s

File: styles/RadioButtons.module.css (hunk 7 above)
Comment (amy-corson-ibigroup) [reply styles/RadioButtons.module.css]: I thought this was a little more of a standard focus indicator than the dashed outline. Open to alternative ideas!

Comment (miles-grant-ibigroup) [reply styles/RadioButtons.module.css]: I think coloring the outline is more standard than adding a second layer

File: components/Alert.tsx [hunk 12]
```diff
@@ -12,7 +12,11 @@ const Alert = ({ alertText, show }: AlertProps) => {
     ? styles.wrapper
     : styles.wrapper + ' ' + styles.hidden
   return (
-    <div aria-hidden={!show} className={wrapperStyles}>
+    <div
+      aria-hidden={!show}
+      className={wrapperStyles}
+      style={{ display: show ? '' : 'none' }}
```
Comment (miles-grant-ibigroup) [inline components/Alert.tsx]: I think this was intentionally not done to improve the animation

File: pages/index.tsx [hunk 13]
```diff
@@ -93,7 +98,7 @@ const Home: NextPage = ({ questions, socketServerUrl }: Props) => {
       {surveyOver && (
         <div className={styles.question}>
           <h1>{t('index.completion')}</h1>
-          <h2>{t('index.completionSecondary')}</h2>
+          <h2>{t('index.completionSubtitle')}</h2>
```
Comment (miles-grant-ibigroup) [inline pages/index.tsx]: This needs to be hidden in the case that the key doesn't exist!

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Looking really good! Tried to find a way to get the grid looking right, but to no success. Just a few nitpicks, and also the :focus border. But looking great overall!

File: styles/RadioButtons.module.css [hunk 14]
```diff
@@ -1,13 +1,14 @@
 .container {
-  display: flex;
-  flex-wrap: wrap;
-  flex-direction: row;
-  gap: 20px;
+  align-items: center;
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
```
Comment (miles-grant-ibigroup) [inline styles/RadioButtons.module.css:4]: This is still not quite working right for me, but I think it's fine for now!

**Review by miles-grant-ibigroup — APPROVED** [review]
Looking really clean! Thanks for all your work on this

**Pull request merged**